torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -75,14 +75,10 @@ def step(self):
         Performs a single optimization step (parameter update).
         """
         self.optim.step()
-        for param_group in self.param_groups:
-            for params in param_group["params"]:
-                if params.grad is None:
-                    continue
-                self.averager.average_parameters(iter(params))
+        self.averager.average_parameters(params=self.param_groups)
 
-    def zero_grad(self):
-        self.optim.zero_grad()
+    def zero_grad(self, set_to_none: bool = False):  # type: ignore[override]
```

Comment:
I am neutral about this change:

- I can understand that this additional arg can help lower memory footprint and may certain behaviors.
- I double checked the other optimizer implementations in fairseq, which is another popular Facebook PyTorch repo, but couldn't find any `zero_grad` method that adds this arg: https://github.com/pytorch/fairseq/search?q=def+zero_grad

Reply:
check nvidia example code.
https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/Classification/ConvNets/image_classification/training.py#L154
i change this only because  i would like to expose this set_to_none argument.
otherwise the input argument of  base Optimizer zero_grad and PostLocalSGDOptimizer is incompatible

Reply:
Sounds good! Thanks for pointing to this example!

Reply:
fix, please cr again

Reply:
Looks good. Thanks!